### PR TITLE
Added grade distribution displaying to SectionSelect

### LINKED ISSUE #163: Design/implement grade distribution display for course select cards
Now that we have grade distributions scraped, we need to design and implement the component that'll be displayed next to professor's names in the course select card, so students can quickly see what the professor's grade distributions were for past semesters.

Should also show what the average GPA was next to it.

### PR DESCRIPTION
**Pull request by gannonprudhomme**
## Description

This adds grade distribution displaying to the `SectionSelect` for each instructor group. It also adds the counting of the total amount of sections to `GradesManager.instructor_performance`, so we can display the `From X total sections` text when the GPA is hovered over (thanks for the idea www.aggiegrades.com)

## How to test

Not sure if there are any edge cases that I can think of. I've just been testing it with `CSCE 121- 202031`, which is my go to typically. 

## Screenshots

|1080p|1440p|
|--|--|
|chrome_PU3RrvU9Zu|chrome_ZRthy0cd3G|

## Related tasks

Closes #163

### CODE REVIEW

**Comment by firejake308** [comment]
How are you deciding the width? I think it would look better if it ended at the same place as the meeting times

**Comment by gannonprudhomme** [comment]
> How are you deciding the width?

Pretty weird way of doing it honestly, the parent div has `90%` and the actual grade dist has `50%` lol. Really just tried to make it look good on my screen and left it at that 😛 . Lmk what a more consistent way would be, as I imagine that `50%` width is gonna have a bad result on your smaller devices

> I think it would look better if it ended at the same place as the meeting times

Honestly how we have it set up currently I'm not sure we can do that right now, although lmk if you have any suggestions. We might have to wait until #238

**Comment by gannonprudhomme** [comment]
Also just pushed a temp commit that fixes the grade dists appearing on the next line. It's added by #251 I think, so I'll drop it once that gets merged & I rebase this

File: autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/SectionSelect/GradeDist/GradeDist.css [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    /* hide the overflow so it actually applies the border-radius on the children */
+    overflow: hidden;
+
+    padding: 0;
+    margin-left: 10px;
+    margin-right: 5px;
+}
+
+.gpa-underline {
+    border-bottom: 1px solid rgb(221, 221, 221);
```
Comment (firejake308) [inline autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/SectionSelect/GradeDist/GradeDist.css:23]: Nice! I wouldn't have known to hover over the GPA unless it had been underlined

**Comment by gannonprudhomme** [comment]
As a reminder for me, here's what we decided from the meeting:

- Right align with show remaining seats?
- Have something to show if there aren’t any grades
    - Gray bar that’s like the grade distribution

**Comment by gannonprudhomme** [comment]
Updated with the above comments, here's how it looks now:

|Normally|When rebased with `frontend/show-remaining-seats`|
|--|--|
|chrome_lLtogMjByP|chrome_8BcpI3yLrl|

**Comment by ryan-conn** [comment]
Rebase this with master whenever you have time now that show remaining seats has been merged, then I'll review it

**Comment by gannonprudhomme** [comment]
Attempted to make the grade widths more "deterministic". Basically just makes it grow until it reaches 10rem. I chose 10rem since it was the width that makes it align with the `No grades available` text (at least on 1080p). Likewise, with this way on smaller screen's it'll take up the entire thing.

Frankly I just want this merged, and think we can address any further width-determining stuff in a later PR when we address the rest of the SectionSelect layout

|1080p|1440p|
|--|--|
|chrome_yCdVmRGRMM|chrome_9HGZ2FUOVT|
